linera-execution/tests/contract_runtime_apis.rs

```diff
@@ -950,3 +950,269 @@ async fn test_perform_http_request(authorized_apps: Option<Vec<()>>) -> Result<(
+            assert_eq!(blob_id2, expected_blob_id2);
+            assert_ne!(blob_id1, blob_id2); // Should be different blobs
+
+            Ok(vec![])
+        },
+    ));
+    application.expect_call(ExpectedCall::default_finalize());
+
+    let context = create_dummy_operation_context(chain_id);
+    let mut controller = ResourceController::default();
+    let operation = Operation::User {
+        application_id,
+        bytes: vec![],
+    };
+
+    let mut tracker = TransactionTracker::new_replaying_blobs(blobs);
+    let result = view
+        .execute_operation(context, operation, &mut tracker, &mut controller)
+        .await;
+
+    assert!(result.is_ok());
+
+    // Verify both blobs were created and tracked
+    let created_blobs = tracker.created_blobs();
+    assert!(created_blobs.contains_key(&expected_blob_id1));
+    assert!(created_blobs.contains_key(&expected_blob_id2));
+
+    let created_blob1 = created_blobs.get(&expected_blob_id1).unwrap();
+    let created_blob2 = created_blobs.get(&expected_blob_id2).unwrap();
+    assert_eq!(created_blob1.bytes(), test_data1);
+    assert_eq!(created_blob2.bytes(), test_data2);
+
+    Ok(())
+}
+
+/// Tests creating an empty data blob.
+#[test_log::test(tokio::test)]
+async fn test_create_empty_data_blob() -> anyhow::Result<()> {
+    let description = dummy_chain_description(0);
+    let chain_id = description.id();
+    let mut view = SystemExecutionState::new(description).into_view().await;
+    let (application_id, application, blobs) = view.register_mock_application(0).await.unwrap();
+
+    let empty_data = vec![];
+    let expected_blob = Blob::new_data(empty_data.clone());
+    let expected_blob_id = expected_blob.id();
+
+    application.expect_call(ExpectedCall::execute_operation(
+        move |runtime, _operation| {
+            let blob_id = runtime.create_data_blob(empty_data.clone()).unwrap();
+            assert_eq!(blob_id, expected_blob_id);
+            Ok(vec![])
+        },
+    ));
+    application.expect_call(ExpectedCall::default_finalize());
+
+    let context = create_dummy_operation_context(chain_id);
+    let mut controller = ResourceController::default();
+    let operation = Operation::User {
+        application_id,
+        bytes: vec![],
+    };
+
+    let mut transaction_tracker = TransactionTracker::new_replaying_blobs(blobs);
+    let result = view
+        .execute_operation(
+            context,
+            operation,
+            &mut transaction_tracker,
+            &mut controller,
+        )
+        .await;
+
+    assert!(result.is_ok());
+
+    // Verify the empty blob was created and tracked
+    let created_blobs = transaction_tracker.created_blobs();
+    assert!(created_blobs.contains_key(&expected_blob_id));
+    let created_blob = created_blobs.get(&expected_blob_id).unwrap();
+    assert!(created_blob.bytes().is_empty());
+
+    Ok(())
+}
+
+/// Tests the contract system API to publish a module with WASM runtime.
```

Comment:
Same here. It feels like `test_publish_module_different_bytecode` covers everything this test does.